src/nvim/path.c

```diff
@@ -2080,8 +2080,14 @@ static int path_get_absolute_path(const char_u *fname, char_u *buf, int len, int
   // expand it if forced or not an absolute path
   if (force || !path_is_absolute_path(fname)) {
     if ((p = vim_strrchr(fname, '/')) != NULL) {
-      STRNCPY(relative_directory, fname, p-fname);
-      relative_directory[p-fname] = NUL;
+      if (p == fname) {
+        // relative to root
```

Comment:
I didn't explain well, this path is executed if there's only one component AND it's relative to root